src/server.rs

```diff
@@ -120,6 +132,17 @@ fn last_unused_address(wallet: &Wallet<AnyBlockchain, Tree>) -> Result<Address,
     wallet.get_address(LastUnused)
 }
 
+fn is_my_address(
+    wallet: &Wallet<AnyBlockchain, Tree>,
+    addr: &str,
+) -> Result<bool, simple_server::Error> {
+    wallet.sync(log_progress(), None).map_err(|_| gen_err())?;
```

Comment:
`sync` is a heavy operation, we should check if the address `wallet.is_mine` first, and only if false call the sync and then the `wallet.is_mine` again.
This is still dos-able, but much better in the best/most common case